cmd/oras/root/cp.go

```diff
@@ -191,3 +196,41 @@ func runCopy(ctx context.Context, opts copyOptions) error {
 
 	return nil
 }
+
+// RecursiveCopy copies an artifact and its referrers from one target to another.
+// If the artifact is a manifet list or index, its predecessor's referrers are copied as well.
+func RecursiveCopy(ctx context.Context, src oras.ReadOnlyGraphTarget, dst oras.Target, dstRef string, root ocispec.Descriptor, opts oras.ExtendedCopyOptions) error {
```

Comment:
@qweeah Let's discuss offline on the `RecursiveCopy` so that we can do better concurrency control.

Reply:
The new recursive copy will be implemented via adding virtual links from referrers of child manifests to index, see dashed links in below graph:
```mermaid
graph TD
Z>v1] -----> index
A["artifact(linux/s390x)"] -- subject --> B["image"]
C["artifact(linux/amd64)"] -- subject --> D["image"]
E["artifact(linux/arm64)"] -- subject --> F["image"]

index -- linux/s390x --> B
index -- linux/arm64 --> D
index -- linux/amd64 --> F

A-.->index
C-.->index
E-.->index
```